[Docs] Add stitching reference (#511)
artsy/convection: docs/schema-stitching.md

# Schema Stitching

Convection takes advantage of [schema stitching](https://artsy.github.io/blog/2018/12/11/GraphQL-Stitching/), which allows us to localize our GraphQL code at the service level, rather than writing everything in Metaphysics. However, there are some steps involved in ensuring that the changes made here propagate correctly throughout our application eco-system.

### Adding or modifying Convection GraphQL schema

1. Make change to convection GraphQL schema, open PR and merge, and be sure its fully deployed to staging

> 1. In Metaphysics, open PR and run `yarn dump-remote-schema`. This will fetch Convection's schema and do some stitching magic. Merge PR, and make sure it gets to staging

Reviewer: ```suggestion
1. In Metaphysics, run `yarn dump-remote-schema` and open a PR. This script fetches Convection's staging schema and does some stitching magic. Merge PR, and make sure it gets to staging.
```
